### PR TITLE
add initial charm skeleton

### PR DESCRIPTION
**Pull request by ca-scribner**
Adds the boilerplate structure of the repo with a stub charm.py and executable but failing tests.

Out of scope: a working istio-core charm.

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: istio-core
```
Comment (dstathis) [inline metadata.yaml:4]: This probably needs to be istio-core-k8s

Comment (dstathis) [reply metadata.yaml:4]: Actually I think the name we came up with was "istio" not "istio-core" so this should be "istio-k8s".

**Review by IbraAoad — DISMISSED** [review]
We usually have `k8s` in the names of the charms specifically designed to only work for kubernetes, meaning this should be called "istio-core-k8s-operator".

Not sure as well if we'll go with the merged `charmcraft.yaml` or leave it split.

File: metadata.yaml (hunk 1 above)
Comment (IbraAoad) [reply metadata.yaml:4]: it's left as "istio-core" in lucid and I think `core` is a bit more indicative that it's not the charm you'll be relating to?, anyways I'm fine with both :D

**Comment by dstathis** [comment]
> We usually have `k8s` in the names of the charms specifically designed to only work for kubernetes, meaning this should be called "istio-core-k8s-operator".
> 
> Not sure as well if we'll go with the merged `charmcraft.yaml` or leave it split.

I reviewed out decision and it should be istio-k8s.

File: metadata.yaml (hunk 1 above)
Comment (ca-scribner) [reply metadata.yaml:4]: renamed to istio-k8s, and also merged the metadata.yaml and charmcraft.yaml files. 

Will put in a request to rename the repo in future.

Comment (ca-scribner) [reply metadata.yaml]: oh wait, didn't even need IS I could rename it myself.  nice

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
```
Comment (IbraAoad) [inline tests/integration/test_charm.py:15]:
```suggestion
METADATA = yaml.safe_load(Path("./charmcraft.yaml").read_text())
```

**Review by IbraAoad — APPROVED** [review]

**Review by dstathis — DISMISSED** [review]

**Review by IbraAoad — APPROVED** [review]

**Pull request merged**